Showing download count of extensions, make empty rating stars less attractive

yiningwang11 opened the pull request:
fix #732

amvanbaren commented:
@deboer-tim Does this PR fix issue #732?

amvanbaren commented:
This is what the changes look like:
![search-downloads](https://github.com/eclipse/openvsx/assets/14129099/d4177839-f300-472e-b559-b70ca089bef8)

amvanbaren commented:
@deboer-tim Is #732 inspired by how the VSCode Marketplace displays search results?
![search-downloads-marketplace](https://github.com/eclipse/openvsx/assets/14129099/601a77bd-903f-4d9a-9c36-176300cf4045)

deboer-tim commented:
The changes look good in the first screenshot, but you can still see why I opened the issue: a sea of no ratings. My preference would still be to remove the stars entirely from the main page, or hide the stars from the main page when there is no rating.

For the ones with 0 downloads it's maybe worse b/c you can't tell if `(0)` means no ratings or no downloads (or both)? I would recommend always keeping the k for downloads (i.e. `(0k)`).

Yes, I did look at VS Code marketplace after opening this, and at the point I looked the bottom line wasn't there - i.e. they weren't showing the ratings at all on the main page and there was no 'free' label.

yiningwang11 commented:
Hello, 
I modified the empty stars icon so it now looks like this:
<img width="211" alt="image" src="https://github.com/eclipse/openvsx/assets/108029327/97c81c91-0be8-4252-bd29-ae648734b32b">
For the download counts, I added a download icon before it so it is more straight forward. It is also hidden on the main page for those extensions without any downloads.
The only problem that I'm having now is that I cannot change the color of the half star icon. I am having problem finding half star icon in the materialui icon libraries with the right part of the star to be grey.

amvanbaren commented:
> I am having problem finding half star icon in the materialui icon libraries with the right part of the star to be grey.

I think you need to add a gray `StarIcon` behind the `StarHalfIcon` to make that work. Give it a try, but removing the stars is fine too.

The download icon makes sense and looks nice. I think you can drop the braces around the download number.

amvanbaren commented:
![742-ratings](https://github.com/eclipse/openvsx/assets/14129099/ccf5810c-18a1-45e1-8270-27a85474ceeb)

Looks good @yiningwang11! @deboer-tim what do you think?

deboer-tim commented:
Sure, looks good.
